shared/middleware/helpers/table.rb

```diff
@@ -140,12 +232,47 @@ def initialize(channel_id, storage_id, table_name)
     @table_name = table_name
 
     @hash = "#{@channel_id}:#{@table_name}:#{@storage_id}"
+    @metadata_hash = "#{@channel_id}:#{@table_name}:#{@storage_id}:metadata"
   end
```

Comment:
It seems like you could use the same `@hash` for both tables. Can you articulate the reasons for using a different `@metadata_hash`? I don't necessarily think it's a bad idea, but it would be helpful to have the reasons recorded.